File: library/src/main/java/com/pengrad/telegrambot/request/EditUserStarSubscription.java
package com.pengrad.telegrambot.request;

import com.pengrad.telegrambot.response.BaseResponse;


public class EditUserStarSubscription extends BaseRequest<EditUserStarSubscription, BaseResponse> {

    /**
     *
     * @param userId Identifier of the user whose subscription will be edited
     * @param telegramPaymentChargeId Telegram payment identifier for the subscription
     * @param isCanceled Pass True to cancel extension of the user subscription;
     *                    the subscription must be active up to the end of the current subscription period.
     *                    Pass False to allow the user to re-enable a subscription
     *                    that was previously canceled by the bot.
     */
    public EditUserStarSubscription(Long userId, String telegramPaymentChargeId, boolean isCanceled) {
        super(BaseResponse.class);
        add("user_id", userId).add("telegram_payment_charge_id", telegramPaymentChargeId).add("is_canceled", isCanceled);
    }
}

File: library/src/main/java/com/pengrad/telegrambot/request/EditUserStarSubscription.kt
package com.pengrad.telegrambot.request

import com.pengrad.telegrambot.response.BaseResponse
import com.pengrad.telegrambot.utility.kotlin.requestParameter

class EditUserStarSubscription(
    userId: Long,
    telegramPaymentChargeId: String,
    isCanceled: Boolean
) : KBaseRequest<EditUserStarSubscription, BaseResponse>(BaseResponse::class) {

    val userId: Long by requestParameter(userId)
    val telegramPaymentChargeId: String by requestParameter(telegramPaymentChargeId)
    val isCanceled: Boolean by requestParameter(isCanceled)

}